### PR TITLE
Extended animals route

### LINKED ISSUE #136: Extending animal route
**Describe the feature**
Add the remaining `category` in the animal's route and fetching data about a particular type of animal.

**Describe the solution you'd like** :zap:

- [ ] add following categories- `bird`, `reptile`, `fish`.
- [ ] add endpoint `animals/:category?type=abc`

**Are you working on this feature? (Yes)**

### PR DESCRIPTION
**Pull request by ritik307**
# Related Issue

-   closes #136 

# Proposed Changes
- [x] add following categories- `bird`, `reptile`, `fish`.
- [x] add endpoint `animals/:category?type=abc`

# Checklist

-   [x] Tests
-   [x] Translations
-   [x] Documentation

### CODE REVIEW

**Comment by ritik307** [comment]
@paramsiddharth @ShauryaAg Plz review.

File: src/api/routes/animals.ts [hunk 1]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
-    } 
+    }
 });
 
 router.get("/:category", (req: Request, res: Response, next: NextFunction) => {
-    const category:any = req.params.category ;
-    const result = data.data.filter((animal:any)=>animal.category === category);
+    const category: any = req.params.category;
+
+    let result = data.data.filter((animal: any) => animal.category === category)[0];
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:21]: Use `.find` method instead of `.filter`

File: src/api/routes/animals.ts [hunk 2]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
+    console.log(result);
     try {
-        if(result.length)
-            res.status(200).json(result);
-        else
-            res.status(404).json({ message: "Invalid category" });    
-    }
-    catch (err) {
+        if (result) {
+            if (Object.keys(req.query).length) {
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:25]: These if statements can be combined into one.

File: src/api/routes/animals.ts [hunk 3]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
+                const type: any = req.query.type;
+                if (!type) {
+                    res.status(404).json({ message: "Invalid query" });
+                } else {
+                    result = result.types.filter((animal: any) => animal.type === type);
+                    if (result.length) 
+                        res.status(200).json(result);
+                    else 
+                        res.status(204).json({ message: "No data found" });
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:34]: Use braces for `if/else` statements.

File: src/api/routes/animals.ts [hunk 4]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
+                }
+            } 
+            else 
+                res.status(200).json(result);
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:38]: Add braces.

File: src/api/routes/animals.ts [hunk 5]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
+        } 
+        else 
+            res.status(404).json({ message: "Invalid category" });
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:41]: Add braces

File: src/api/routes/animals.ts [hunk 6]
```diff
@@ -7,29 +7,43 @@ const router = express.Router();
...
         });
-    } 
+    }
 });
 
 router.get("/:category", (req: Request, res: Response, next: NextFunction) => {
-    const category:any = req.params.category ;
-    const result = data.data.filter((animal:any)=>animal.category === category);
+    const category: any = req.params.category;
+
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:20]: Remove extra line.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by ritik307** [comment]
@ShauryaAg Done! Plz check

File: src/api/routes/animals.ts [hunk 7]
```diff
@@ -7,29 +7,46 @@ const router = express.Router();
...
+        if (result && Object.keys(req.query).length) {
+            const type: any = req.query.type;
+            if (!type) {
+                res.status(404).json({ message: "Invalid query" });
+            } else {
+                result = result.types.filter((animal: any) => animal.type === type);
+                if (result.length) {
+                    res.status(200).json(result);
+                } 
+                else {
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:32]: Move `else` to upper line, like 
```js
if {
...
} else {
...
}
```

File: src/api/routes/animals.ts [hunk 8]
```diff
@@ -7,29 +7,46 @@ const router = express.Router();
...
+                    res.status(204).json({ message: "No data found" });
+                }
+            }
+        } 
+        else {
+            if (result) {
+                res.status(200).json(result);
+            } 
+            else {
+                res.status(404).json({ message: "Invalid category" });
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:42]: Move `else` to upper line

File: src/api/routes/animals.ts [hunk 9]
```diff
@@ -7,29 +7,46 @@ const router = express.Router();
...
+                result = result.types.filter((animal: any) => animal.type === type);
+                if (result.length) {
+                    res.status(200).json(result);
+                } 
+                else {
+                    res.status(204).json({ message: "No data found" });
+                }
+            }
+        } 
+        else {
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:37]: Move `else`

File: src/api/routes/animals.ts [hunk 10]
```diff
@@ -7,29 +7,46 @@ const router = express.Router();
...
-    } 
+    }
 });
 
 router.get("/:category", (req: Request, res: Response, next: NextFunction) => {
-    const category:any = req.params.category ;
-    const result = data.data.filter((animal:any)=>animal.category === category);
+    const category: any = req.params.category;
+    let result = data.data.find((animal: any) => animal.category === category);
+    console.log(result);
```
Comment (ShauryaAg) [inline src/api/routes/animals.ts:21]: Remove unnecessary console logs.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Review by ShauryaAg — APPROVED** [review]

**Comment by ritik307** [comment]
@ShauryaAg Plz add appropriate tags

**Pull request merged**